rust/hyperlane-core/src/traits/pending_operation.rs

```diff
@@ -129,7 +134,35 @@ pub enum PendingOperationStatus {
     Confirm(ConfirmReason),
 }
 
-#[derive(Display, Debug, Clone)]
+impl Encode for PendingOperationStatus {
+    fn write_to<W>(&self, writer: &mut W) -> std::io::Result<usize>
+    where
+        W: Write,
+    {
+        // Serialize to JSON and write to the writer, to avoid having to implement the encoding manually
+        let serialized = serde_json::to_vec(self)
+            .map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, "Failed to serialize"))?;
+        writer.write(&serialized)
+    }
+}
+
+impl Decode for PendingOperationStatus {
+    fn read_from<R>(reader: &mut R) -> Result<Self, HyperlaneProtocolError>
+    where
+        R: std::io::Read,
+        Self: Sized,
+    {
+        // Deserialize from JSON and read from the reader, to avoid having to implement the encoding / decoding manually
+        serde_json::from_reader(reader).map_err(|_| {
```

Comment:
is there any way to get err in what's returned / maybe surface what the problematic bytes were?